### DISCUSSION TITLE
There is any tapping gesture on TapDown instead of TapUp and hold gesture?

### DISCUSSION THREAD

**Discussion by PGama-Rodrigo** [dotnet/maui#20309] [Q&A] [answered]
Hello. I'm new on .NET MAUI and I find it wonderful on many ways! But I just can't find anything that provides me a TappedGesture Event Handler when the tap is initialized iastead on completion (TapDown instead of TapUp).

I also can't find anything for hold tap gestures. I was hopping for this to implement different kinds of control on my application.

I'm looking around for any solution that does the job, being native or by entensions, but I found nothing so far.

Can you help me with ideas or suggestions?

**Comment by bradencohen** [comment]
Would a Pointer Gesture work? https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/gestures/pointer?view=net-maui-8.0&tabs=macios

**Comment by JamesBrooksbankIceland** [comment, marked as answer]
I recently needed to use a hold gesture for a project we are working on in Maui, theres nothing native and the community toolkit which did have these gestures built in, no longer have them. However, i found a library which used that implementation as a basis within Maui,
https://github.com/felipebaltazar/TouchEffect.Maui

We implemented it and the long gesture works, there are multiple other commands in there. The long gesture definitely works in .Net 8 which is what our project is developed in.
